core/node/node_sync/src/external_io.rs

```diff
@@ -341,10 +342,13 @@ impl StateKeeperIO for ExternalIO {
             .await?
             .protocol_versions_dal()
             .save_protocol_version(
-                protocol_version
-                    .version_id
-                    .try_into()
-                    .context("cannot convert protocol version")?,
+                ProtocolSemanticVersion {
+                    minor: protocol_version
+                        .version_id
+                        .try_into()
+                        .context("cannot convert protocol version")?,
+                    patch: VersionPatch(0),
```

Comment:
Just in case, what is the behavior when on stage we release 0.25.0, but then we realize that it was bad and use 0.25.1 rightaway on testnet. It will mean that the data will be different in ENs and will believe that the provided data is actually for 0.25.0, while it was for 0.25.1

Reply:
so ENs will just have some vks and only one patch per minor version, they don't use them so it should be fine